examples/failing/ProgrammableTypeErrorsTypeString.purs

```diff
@@ -0,0 +1,18 @@
+-- @shouldFailWith NoInstanceFound
+
+module Main where
+
+import Prelude
+import Control.Monad.Eff (Eff)
+import Control.Monad.Eff.Console (log)
+
+newtype MyType a = MyType a
+
+instance cannotShowFunctions :: Fail ("Don't want to show" <> TypeString (MyType a) <> "because.") => Show (MyType a) where
```

Comment:
That would be cool indeed :). But that would need changes to the parser and the AST I think. Maybe we keep that for a seperate PR?